tools/ds-identify

```diff
@@ -836,9 +836,11 @@ dscheck_NoCloud() {
     local fslabel="cidata CIDATA" d=""
     case " ${DI_KERNEL_CMDLINE} " in
         *\ ds=nocloud*) return ${DS_FOUND};;
+        *\ ci.ds=nocloud*) return ${DS_FOUND};;
```

Comment:
I know ds-identify will early exist with  `debug 1 "datasource '$DI_DSNAME' specified."`                       
 it sees `ci.ds=openstack` and our `cloudinit/sources/__init__py: override_ds_detect` in python will return True for _check_and_get_data. Are there any other elements we need to cover to ensure that the processed cfg_list of potential datasources from `find_source` is limited to just a single entry, instead walking through potentially 20 datasources?

Reply:
> so we've added ci.ds support ci.ds=nocloud but aren't we instead missing an element to allow ds-identify to generally handle lowercase datasource name on /proc/cmdline?

Good point, I don't think we need to add this here if we're going to prefer `ds=nocloud` (since it isn't already supported by ds-id), and lowercase match is required since we want parity between ds-id and the Python parsing code, so I'll implement that.